### PR TITLE
Added some missing functionality

### PR DESCRIPTION
**Pull request by BenDol**
- Added some missing lua methods.
- Added creature move locking (setMoveLocked/isMoveLocked).
- Added less loss attribute to vocations to customize death loss per vocation.
- Added definitions.h to cmake source definitions.
- Always reload global.lua libraries.

Will squish the commits when its ready.

### CODE REVIEW

File: src/creature.h [hunk 1]
```diff
@@ -182,6 +182,13 @@ class Creature : virtual public Thing
 			hiddenHealth = b;
 		}
 
+		bool isMoveLocked() const {
+			return moveLocked;
+		}
+		void setMoveLocked(bool ml) {
```
Comment (WibbenZ) [inline src/creature.h]: ml should be named something else

File: src/game.cpp [hunk 2]
```diff
@@ -3115,6 +3115,16 @@ void Game::playerTurn(uint32_t playerId, Direction dir)
 
 	player->resetIdleTime();
 	internalCreatureTurn(player, dir);
+
+	if(player->getDirection() != dir || (player->getAccountType() < ACCOUNT_TYPE_GAMEMASTER))
```
Comment (WibbenZ) [inline src/game.cpp]: Curly brackets, "if (" not "if(" note the space.

File: src/vocation.cpp [hunk 3]
```diff
@@ -109,6 +109,10 @@ bool Vocations::loadFromXml()
 			voc.fromVocation = pugi::cast<uint32_t>(attr.value());
 		}
 
+		if ((attr = vocationNode.attribute("lessloss"))) {
```
Comment (WibbenZ) [inline src/vocation.cpp]: Well you added the xml loading part, but nothing happens with the code.
Did you miss uploading some code?

**Comment by PrinterLUA** [comment]
GM teleport feature can be achived through lua:
https://otland.net/threads/event-walkthrough-everything-ctrl-arrow-keys-tfs-1-x.226973/

**Comment by vankk** [comment]
I agree with @PrinterLUA .

File: src/vocation.cpp (hunk 3 above)
Comment (BenDol) [inline src/vocation.cpp]: Appears I forgot to copy something, will do it now.

File: src/creature.h (hunk 1 above)
Comment (BenDol) [inline src/creature.h]: I disagree, unless you have a better name this is perfectly fine.

**Comment by BenDol** [comment]
It should be added to the tfs project none the less.

**Comment by ranisalt** [comment]
> Added definitions.h to cmake source definitions.

Added **header** to **source**... are you sure? It's not needed :P

**Comment by BenDol** [comment]
When run your project using cmake IDE's like qtcreator only load the files from the cmake definition.

**Comment by ranisalt** [comment]
I use CLion, I have no problems and didn't understand what you mean. Anyway it's your IDE's fault, not TFS'.

**Comment by BenDol** [comment]
It compiles fine, it just doesn't include definitions.h in the file list for some IDE's. Its not a big deal.

**Comment by ranisalt** [comment]
Do the same happens with enums and const headers?

**Comment by BenDol** [comment]
If its not in the cmake resource files then yes they won't be imported to the project. Since the IDE uses cmake to build the source file index.

File: src/luascript.cpp [hunk 4]
```diff
@@ -7059,6 +7058,18 @@ int LuaScriptInterface::luaCreatureSetMaxHealth(lua_State* L)
 	return 1;
 }
 
+int LuaScriptInterface::luaCreatureIsHealthHidden(lua_State* L)
```
Comment (djarek) [inline src/luascript.cpp]: Is there any particular reason you moved this function?

Comment (conde2) [inline src/luascript.cpp]: He just grouped it with luaCreatureSetHiddenHealth

File: src/vocation.cpp (hunk 3 above)
Comment (ranisalt) [inline src/vocation.cpp]: `deathpenaltyreduction` instead of `lessloss` sounds better

Comment (BenDol) [inline src/vocation.cpp]: I thought about that too but its very long winded

**Comment by BenDol** [comment]
Don't have time to finish this PR, will leave it open for the ideas at least.

**Comment by nekiro** [comment]
Closing this as it's not maintained and deprecated.

**Pull request closed without merging**